addons/calendar/models/discuss_channel_member.py

```diff
@@ -0,0 +1,13 @@
+from odoo import fields, models
+
+
+class DiscussChannelMember(models.Model):
+    _inherit = "discuss.channel.member"
+
+    def _should_invite_members_to_join_call(self):
+        if self.channel_id.calendar_event_id:
+            return (
+                super()._should_invite_members_to_join_call()
+                and fields.Datetime.now() > self.channel_id.calendar_event_id.start
```

Comment:
Also there is currently nothing (unique constraint) to prevent having the same `channel_id` in multiple events, which in this case would crash this line when accessing `start`.

Reply:
I added a unique constrain on `videocall_channel_id`.

Reply:
videocall_channel_id is shared when you have recurring events. It prevent creating hundred of channel and it allow to reuse channels when you meet regularly.  Once the event is passed, the channels are automatically vacuumed after some time ? Otherwise, it will probably bloat the database as recurring meeting can create up to 720 events. It will also force to reshare a new URL each time the event is joined instead of keeping a bookmark.

Reply:
Thank you for the feedback! We'll keep it non-unique then.

How should the spec of the current task work for them? When is it supposed to be allowed to ring for recurring event? Always? Never? Some complex condition based on today's recurring event start date + expected event duration?

cc @aju-odoo

Reply:
I would keep it simple and allow to use the channel all the time but I am not familiar with the flows you are facing with discuss.

Reply:
> I would keep it simple and allow to use the channel all the time but I am not familiar with the flows you are facing with discuss.

The current issue we are trying to solve is that curious guest that follow the meeting link before the start date are going to call the others.

@aju-odoo What about calling only between the start (minus 5/10 min for early birds?) and the end of the event? I think an old link should also not call the user since the meeting is supposed to be over.

Reply:
I think we should keep things simple and never ring. Users receive a reminder for the event anyway.